bin/pyenv-installer

```diff
@@ -50,7 +61,57 @@ checkout "${GITHUB}/pyenv/pyenv-update.git"     "${PYENV_ROOT}/plugins/pyenv-upd
 checkout "${GITHUB}/pyenv/pyenv-virtualenv.git" "${PYENV_ROOT}/plugins/pyenv-virtualenv"  "master"
 checkout "${GITHUB}/pyenv/pyenv-which-ext.git"  "${PYENV_ROOT}/plugins/pyenv-which-ext"   "master"
 
+write_source() {
+  # expand ~ to user's home
+  local target="${1/#\~/$HOME}"
+
+  echo "Update file: $1" >&2
+
+  echo "" >>$target
+  echo '# pyenv' >>$target
+  echo 'export PATH="'"$PYENV_ROOT/bin:"'$PATH"' >>$target
+  echo 'eval "$(pyenv init -)"' >>$target
+  echo 'eval "$(pyenv virtualenv-init -)"' >>$target
+
+  USER_PATH_ADDED=true
+}
+
```

Comment:
I still heavily dislike the duplication with `pyenv-init`.
It's going to be a constant headache and source of errors during maintenance.

I understand the desire to keep installation logic out of `pyenv-init` as it's of no use after installation. But however you look at it, the bulk of it is already in Pyenv-Init anyway!

Reply:
The "is-Pyenv-shell-setup-already-done" part of the logic is something that could probably be kept out of Pyenv-Init -- if it proves to be different enough from existing Pyenv-Init logic.

Reply:
We _could_ add `pyenv init --profile` and `pyenv init --rc` to output just the stuff that needs to be written, there write it here. But since those files are detected inside `--detect-shell`, it seems simpler and even less code in `pyenv-init` to just add a single `pyenv-init --install`.